.github/workflows/ci.yml

```diff
@@ -61,6 +61,36 @@ jobs:
         env:
           RUSTDOCFLAGS: ${{ matrix.rust_channel == 'nightly' && '-Dwarnings --cfg=docsrs' || '-Dwarnings' }}
 
+  check-external-types:
+    name: Validate external types appearing in public API
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout sources
+        uses: actions/checkout@v4
+        with:
+          persist-credentials: false
+      - name: Install rust toolchain
+        uses: dtolnay/rust-toolchain@master
+        with:
+          toolchain: nightly-2023-10-10
+          # ^ sync with https://github.com/awslabs/cargo-check-external-types/blob/main/rust-toolchain.toml
+      - run: cargo install --locked cargo-check-external-types
+      - name: run cargo-check-external-types for rcgen/
+        working-directory: rcgen/
+        run: cargo check-external-types --all-features
+
+  semver:
+    name: Check semver compatibility
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout sources
+        uses: actions/checkout@v4
+        with:
+          persist-credentials: false
+
+      - name: Check semver
+        uses: obi1kenobi/cargo-semver-checks-action@v2
```

Comment:
What exactly does this do? Can this be replaced by a bunch of commands?

Reply:
> What exactly does this do? 

It installs Rust, finds the latest binary release for the backing tool, downloads it, installs it, translates the config from the github action to the tool's arguments, and runs it.

> Can this be replaced by a bunch of commands?

Sure. Are you requesting that change?

Reply:
Hmm maybe it's better to use the external dependency here. If there is any breakage we can just disable it. It's not part of the "CI main path" I guess.